test/gen_source.rb

```diff
@@ -0,0 +1,113 @@
+require 'optparse'
+require 'yaml'
+require 'erb'
+
+
+REGEXT_PRETTY = %r{
+ =\s            # We are interested to the right of the equal sign
+ (
+    ""|         # Empty string
+    ".*?[^\\]"| # String who can contain space and quoted string
+    [^\s,]+     # Anything except space and comma
+ )
+}x
+
+SOURCE_TEMPLATE =  <<-TEXT
+/* Code generaed by source_callbacks_generator.rb */
+
+#include "component.h"
+#include "create.h"
+
+void btx_push_usr_messages(struct common_data_s *common_data) {
+
+    <%- data.each do | entry | -%>
+    <%- entry.fetch(:times,1).times do -%>
+    btx_push_message_<%= entry[:name].gsub(":","_") %>(common_data,<%=  entry[:field_values].join(",") %>);
```

Comment:
It's not the correct sub. Need to transfer a lot of other characters. For example `*,`

Reply:
This is actually not needed,  because your sanitizer `data[:name]=head.gsub(/[^0-9A-Za-z\-]/, '_') # Should reuse metababel mangling`

Reply:
Yes, but this file should be generated by the `log` file, which is not mangled.